mem/interface.md

```diff
@@ -34,14 +34,14 @@ We define the `AbstractByte` type as follows, where `Provenance` will later be i
 
 ```rust
 #[derive(PartialEq, Eq)]
-pub enum AbstractByte<Provenance> {
+pub enum AbstractByte<Provenance: Eq + Hash + Clone> {
```

Comment:
In Rust we usually avoid trait bounds in the types and only have them on the `impl` blocks that need them.

Reply:
Ah right, I've heard that before. What was the rationale for that again?

Reply:
I think the ideas is to generally only state assumptions where we actually need them.